c10/core/StorageImpl.h

```diff
@@ -55,6 +55,11 @@ struct C10_API StorageImpl : public c10::intrusive_ptr_target {
         resizable_(resizable),
         received_cuda_(false),
         allocator_(allocator) {
+    if (data_ptr_ == nullptr) {
```

Comment:
I don't think we want to allow empty dataptr and allocate under the hood here. That is a significant semantic creep for this function

Reply:
> I don't think we want to allow empty dataptr and allocate under the hood here. That is a significant semantic creep for this function

Ezyang suggested “pass in the nullptr and have the constructor deal with it” so I have changed to this. Maybe rollback such code is better?